frontend/src/tests/unit/Api.test.js

```diff
@@ -0,0 +1,51 @@
+import { convertSecondsToStringInfo, getRemainingDate, parseJwt } from '../../utils/Api'
+import { testEndDates } from './testHelpers/endDates'
+import { testInvalidTokens, testProfessorTokens, testStudentTokens } from './testHelpers/tokens'
+
+describe('parseJwt() tests', () => {
+  it.each(testStudentTokens)('student token %s parsed correctly', (token) => {
+    //when
+    const parsedToken = parseJwt(token.access_token)
+
+    //then
+    expect(parsedToken).toEqual(token.token_data)
+  })
+  it.each(testProfessorTokens)('professor token %s parsed correctly', (token) => {
+    //when
+    const parsedToken = parseJwt(token.access_token)
+
+    //then
+    expect(parsedToken).toEqual(token.token_data)
+  })
+  it.each(testInvalidTokens)('invalid token %s throws error', (token) => {
+    //when
+    const parsedToken = parseJwt(token.access_token)
+
+    //then
+    expect(parsedToken).toBeNull()
+  })
+})
+
+describe('getRemainingDate() tests', () => {
+  beforeAll(() => {
+    jest.useFakeTimers().setSystemTime(new Date('2022-05-13T12:00:00.000Z'))
```

Comment:
Jeżeli z tego korzystasz to powinieneś dodać
```js
afterAll(() => {
  jest.useRealTimers()
})
```

Reply:
Nie powinno mieć to żadnego znaczenia bo sam plik testów jest odpalany na oddzielnym wątku i pozostałe testy nie są zależne od timera, ale mogę dodać, czemu nie